how-to-guides/databases/xata.md

```diff
@@ -0,0 +1,322 @@
+
+{% code %}
+```
+psql postgresql://<workspace-id>:$XATA_API_KEY@<region>.sql.xata.sh/<database-name>:<branch>?sslmode=require
+```
+{% endcode %}
+
+![Xata Database Access](../../.gitbook/assets/xata-db-access.png)
+
+{% endtab %}
+
+{% tab title="Xata Shell" %}
+
+## Existing Xata Project
+
+### Step 1: Creating the API Key
+
+First you need to get the API access key so that we can connect to the database:
+
+1. Go to **[Xata Account Settings](https://app.xata.io/settings)**.
+2. In the **Personal API keys** section, click on **Add a key** and enter the name of the key.
+3. Note down the secret API key.
+
+### Step 2: Creating the Recipe for Workspace
+
+You will need to create a **Recipe** in DevZero equiped with the neccesary tools like `node`, `xata` and store your API keys as well:
+
+1. Go to **DevZero** Dashboard and open the **Recipes** tab.
+2. Click on **New Recipe** and enter the name of recipe.
+3. Click on **Create a recipe** and your recipe will be created.
+4. Replace the recipe `yaml` file with the snippet below. Remember to replace the placeholder API key with the Key you noted down in the above steps:
+
+{% code %}
+```
+version: "3"
+
+build:
+  steps:
+    - type: apt-get
+      packages: ["build-essential", "curl", "git", "nano", "software-properties-common", "ssh", "sudo", "tar", "unzip", "vim", "wget", "zip"]
+    - type: command
+      command: |
+        curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash
+        echo 'export NVM_DIR=$HOME/.nvm' | sudo tee -a /etc/profile.d/nvm-installation.sh
+        echo '[ -s $NVM_DIR/nvm.sh ] && \. $NVM_DIR/nvm.sh' | sudo tee -a /etc/profile.d/nvm-installation.sh
+        . /etc/profile.d/nvm-installation.sh
+        nvm install 21.0.0
+        npm install -g @xata.io/cli
+      
+runtime:
+  environment:
+    - name: XATA_API_KEY
+      value: <api-key>
+```
+{% endcode %}
+
+![DevZero Recipe](../../.gitbook/assets/devzero-recipe-yaml.png)
+
+5. After editing the recipe, click on **Save and Build**.
+6. When the build is successful, click on **Publish**.
+7. After the recipe is published, click on **Launch** and create a new workspace by providing a **Workspace Name**.
+8. Click on **Launch** and your workspace will be created.
+
+### Step 3: Connecting to the Database
+
+Follow the below steps to connect to the database instance:
+
+1. Go to DevBox and use the following command to authenticate your xata account:
+
+{% code %}
+```
+xata auth login
+```
+{% endcode %}
+
+2. Choose the `Use an existing API key` option while authenticating, and then hit **Enter/Return** with the API Key shown when prompted.
+3. Use the following command for access the database through Xata shell:
+
+{% code %}
+```
+xata shell
+```
+{% endcode %}
+
+4. Select the **Workspace**, **Database** and **Branch** when prompted.
+
+![Xata Database Shell Access](../../.gitbook/assets/xata-db-shell-access.png)
+
+## New Xata Workspace
```

Comment:
@Sid-Lais fixed most of your issues w/ #132 
i can't figure out if if you want 2 headings w/ `New Xata Workspace` - esp given that they are both at the `##` level

Reply:
This is because there are two sections: one for Xata shell access, and one for xata Postgres access. The two headings are such that the existing and new workspace are for the two sections, respectively.

Reply:
ah i see - do you want to use markdown headings for both the sections?

Reply:
Yes, I want to use the H2 heading for them